### PR TITLE
fixes for EXTERNAL, launch failures

### PR DESCRIPTION
**Pull request by paulswartz**
Correctly failing the Trike-dev deploy: https://github.com/mbta/trike/actions/runs/5269734970

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202390454457132

### CODE REVIEW

File: deploy-ecs/deploy.sh [hunk 1]
```diff
@@ -25,7 +25,7 @@ function check_deployment_complete() {
   deployment_details="${1}"
 
   # get and print current task counts
-  desired_count="$(echo "${deployment_details}" | jq -r '.desiredCount')"
+  desired_count="$(echo "${deployment_details}" | jq -r '[.desiredCount, 1] | max')"
```
Comment (ianwestcott) [inline deploy-ecs/deploy.sh:28]: So this assumes that `desired_count` should always be at least `1`? That could cause spurious failures if we have a service's task count intentionally set to `0`, but that's an edge case.

**Review by krisrjohnson21 — APPROVED** [review]
🧼

File: deploy-ecs/deploy.sh (hunk 1 above)
Comment (paulswartz) [reply deploy-ecs/deploy.sh:28]: If you desire 0 tasks, what is being deployed?

**Comment by paulswartz** [comment]
Tagged v1 and v1.22.

File: deploy-ecs/deploy.sh (hunk 1 above)
Comment (ianwestcott) [reply deploy-ecs/deploy.sh:28]: - A different container image tag
- Changes from the template task definition

Sometimes we have to take a service down to perform maintenance and then bring it back up with a different version / different configuration that accounts for the maintenance. Not a big deal, just something we'll have to keep in mind in that situation.

**Pull request merged**